### ISSUE TITLE
Doesn't work in Win11

### ISSUE THREAD

**Issue by nezorflame** [diddledani/one-script-wsl2-systemd#13]
Hi!

I'm currently unable to use GPG through the bridge.
Basic install of the script with only `-d Ubuntu` flag (so GPG was installed by the script), fresh Ubuntu install.

After restarting WSL, I get the next output:

```shellscript
Failed to connect to bus: No such file or directory
mkdir: cannot create directory ‘/run/user/1000’: Permission denied
2021/10/03 23:35:51 socat[70] E bind(5, {AF=1 "/run/user/1000/gnupg/S.gpg-agent"}, 34): No such file or directory
> 
```

It partly looks like [this error](https://github.com/diddledani/one-script-wsl2-systemd/issues/2#issuecomment-780774196) which I got during the previous run under Win10.

**Comment by diddledani** [comment]
Can you re-run the installation using a fresh copy of the install.ps1 script (I've just pushed some changes)? The files as of now should wait for systemd more reliably so that you aren't logged into the namespace "too early", which is likely the issue here.

There is another issue with the GPG support in that it only works in the first shell you open after restarting the distro. I haven't worked out why it fails on subsequent launches yet.

**Comment by nezorflame** [comment]
```text
Waiting for systemd to finish booting
..
Systemd is ready. Logging in.
Welcome to Ubuntu 20.04 LTS (GNU/Linux 5.10.43.3-microsoft-standard-WSL2 x86_64)
```

🎉

**Comment by nezorflame** [comment]
Right, while this works, actually I have problems with Docker and SSH now.
ssh-agent seems to detect the keys from Windows only on the first session, other sessions give this error:

```text
> ssh-add -l
error fetching identities: communication with agent failed
```

Docker just doesn't detect that it's running, although integration is enabled:

```text
> docker version
The command 'docker' could not be found in this WSL 2 distro.
...
```

**Comment by nezorflame** [comment]
Quick update - I've reset Ubuntu and tried again on a clean system.
Disabled and enabled Docker WSL integration, returned to WSL - `docker` works, but `ssh-add -l` shows the same error.

**Comment by nezorflame** [comment]
After shutting down with `wsl --shutdown` `ssh-add -l` works again:

```
Waiting for systemd to finish booting
..
Systemd is ready. Logging in.
> ssh-add -l
*my keys are shown*
```

**Comment by diddledani** [comment]
thanks for the updates. I've seen this behaviour of ssh-agent and gpg-agents both failing in subsequent sessions where they work fine in the very first session. I'm unsure what's up but I think it's something to do with the WSL interop feature not allowing cross-session calls - i.e. your second session accessing the socat running under the first session, even if you've closed your first session.

**Comment by nezorflame** [comment]
OK, should we create a separate issue for that then? Or just leave it as is here?

**Comment by nezorflame** [comment]
Anything on this?
